### PR TITLE
Add useUploadAttachment hook (#7617)

### LINKED ISSUE #6565: Editing a RICH_TEXT field should update the attachment table when new images are added.
## Scope & Context

Enrich text fields allow you to include images in your text, however images are uploaded through the server and directly inserted in the json without editing the attachment table. We would like to update the table as well to track those attachments and show them in the appropriate "files" tab.

## Technical inputs

TODO

### PR DESCRIPTION
**Pull request by yadavshubham01**
Reusing the useUploadAttachment Hook

In the implementation of the feature to ensure the attachment table is updated whenever new images are added to a RICH_TEXT field #7617 , it is likely that the useUploadAttachment hook is reused.

The useUploadAttachment hook is responsible for handling the upload of attachments, including images, and returning the uploaded file URL. By reusing this hook, you can leverage its existing functionality to handle image uploads within the RICH_TEXT field.

In this case, the modified image handling logic would utilize the useUploadAttachment hook to upload new images added to the RICH_TEXT content. The hook would then return the uploaded file URL, which would be used to update the attachment table with the details of the newly added images.

By reusing the useUploadAttachment hook, you can avoid duplicating code and ensure consistency in the way attachments are handled throughout the application.

Fixes #6565

### CODE REVIEW

**Comment by yadavshubham01** [comment]
@charlesBochet please review this PR

File: packages/twenty-front/src/modules/activities/files/hooks/useUploadAttachment.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    });
+
+    const uploadedFileData = result?.data?.uploadFile;
+
+    // Check if the upload result exists
+    if (!uploadedFileData) {
+      throw new Error("Couldn't upload the image.");
+    }
+
+    // Get the absolute URL of the uploaded file
```
[the remark this answers is not in the document]
Comment (yadavshubham01) [reply packages/twenty-front/src/modules/activities/files/hooks/useUploadAttachment.tsx]: done 👍

**Comment by charlesBochet** [comment]
This PR does not look functional

**Comment by yadavshubham01** [comment]
there any issue in this PR @charlesBochet

**Review by lucasbordeau — APPROVED** [review]

**Comment by lucasbordeau** [comment]
Follow-up issue : https://github.com/twentyhq/twenty/issues/8351

**Pull request merged**